### PR TITLE
Zone registry tests

### PR DESCRIPTION
**Pull request by jessopb**

### CODE REVIEW

**Comment by jessopb** [comment]
Input requested:
 - [ ] Is it necessary to read `tags` for input interchangeably with `data`
 - [ ] Verify that each registry will maintain a roles table to verify any assigns it receives
 - [ ] Verify the return types, specify best practices for responses.
 - [ ] Critique Handler names

**Comment by jessopb** [comment]
> Input requested:

> * [ ]  Verify that each registry will maintain a roles table to verify any assigns it receives

After a user wallet sends an update to a zone process, there were two techniques we considered. 
1. [Wallet] --- updates ---> [Zone] --- assigns to ---> [Primary Registry] --- assigns to --> [Subscribed Registry]
where any Creation or Update Roles action creates a role table on each registry.

2. [Wallet] --- updates ---> [Zone] --- assigns update to ---> [Primary Registry] --- assigns update to --> [Subscribed Registry]
[Zone] --- sends confirmation to ---> [Primary Registry] --- assigns confirmation to --> [Subscribed Registry]
where each registry waits for a match between a wallet that updates data, and a confirmation from the zone to verify it.

**Comment by jessopb** [comment]
> Input requested:
> 
> * [ ]  Is it necessary to read `tags` for input interchangeably with `data`

In previous versions we made sure to allow Profile handlers to parse their input from either data or tags.
For example, currently action Zone.Metadata-Set takes values in msg.data. Is it important to implement that metadata set data in tags as well so that people can optionally use it that way?

**Comment by jessopb** [comment]
Currently the Zone Metadata Registry expects specific data values for the profile, where the values are sqlite columns.


I briefly considered a KV table
id | metadata_key | metadata_value
xyz | "display_name" | "Steve"
xyz | "updated_at" | "12435677"

This would support any zone profile metadata.

File: services/tests/unit/zoneRegistry/zoneMetaRegistry.test.js [hunk 1]
```diff
@@ -0,0 +1,97 @@
+import { test } from 'node:test'
```
Comment (jessopb) [inline services/tests/unit/zoneRegistry/zoneMetaRegistry.test.js:1]: Downstream Zone Metadata Registry code

File: services/zone-registries/zone-metadata-registry.lua [hunk 2]
```diff
@@ -0,0 +1,711 @@
+local json = require('json')
```
Comment (jessopb) [inline services/zone-registries/zone-metadata-registry.lua:1]: Downstream Zone Metadata Registry Code

**Comment by twilson63** [comment]
> Input requested:
> 
> * [ ]  Is it necessary to read `tags` for input interchangeably with `data`
> * [ ]  Verify that each registry will maintain a roles table to verify any assigns it receives
> * [ ]  Verify the return types, specify best practices for responses.
> * [ ]  Critique Handler names

RE: 1, imo tags should classify the message, data should provide the contents of the message.
RE: 2, is it required for a registry to maintain roles? Are there use case for zones where that is not important?
RE: 3, need more info
RE: 4, handler names should use https://specs.g8way.io/?tx=XV8-NMh5uhLQflkrUs_7mcgj33pOWqQno1wyVi__j6w spec as a guide

**Pull request closed without merging**